### DISCUSSION TITLE
query decorator

### DISCUSSION THREAD

**Discussion by aelbore** [lit/lit#2578] [Q&A] [answered]
does it possible to use query decorator to querySelect into page? for example

```javascript
@query('my-modal') modal
```

equivalent to
```javascript
get modal() {
 return document.querySelector('my-modal')
}
```

**Comment by justinfagnani** [comment, marked as answer]
The actual equivalent is:
```ts
get modal() {
 return this.renderRoot?.querySelector('my-modal') ?? null;
}
```
where `this.renderRoot` is usually `this.shadowRoot` unless you've overridden `createRenderRoot()`.

This is documented here: https://lit.dev/docs/components/shadow-dom/#query

**Reply by aelbore** [reply]
Thanks @justinfagnani 
so basically we cannot use query decorator outside shadow root unless we override the createRenderRoot function? 
so its not good practice query element outside shadow root?

**Reply by justinfagnani** [reply]
If you want to query the document, just use `document.querySelector()`, there's nothing wrong with that. `@query` is just a shorthand for `this.shadowRoot.querySelector()` that's all.

**Reply by aelbore** [reply]
i agree thanks 👍
